libs/ledger-state/src/Cardano/Ledger/State/Query.hs

```diff
@@ -36,14 +36,16 @@ import qualified Data.VMap as VMap
 import qualified Data.Vector.Generic as VG
 import qualified Data.Vector.Generic.Mutable as VGM
 import Database.Persist.Sqlite
+import Database.Persist.Class.PersistEntity (SafeToInsert)
 
 -- Populate database
 
 insertGetKey ::
   ( MonadIO m,
     PersistUniqueWrite backend,
     PersistRecordBackend record backend,
-    AtLeastOneUniqueKey record
+    AtLeastOneUniqueKey record,
+    SafeToInsert record
```

Comment:
@nfrisby Hi! :slightly_smiling_face: I was wondering if we could do without this contraint. If we could, we would not need to bump the `persistent < 2.14` constraint present in `cabal.project`.

Reply:
I only put it there to fix an error I was seeing. Might be totally inconsequential now.

In fact, in did a little more dev today on top of a rebase that @amesgen did for me. Looks like he removed it already 👍

Reply:
> Looks like he removed it already

Exactly, as I got compilation errors with the new constraint :+1:

Reply:
Got it! Thanks!